### DISCUSSION TITLE
Directory not empty @ dir_s_rmdir when running livecheck on casks using the `extract_plist` livecheck strategy

### DISCUSSION THREAD

**Discussion by PythonCoderAS** [Homebrew/discussions#3369] [Tap maintenance and Homebrew development] [answered]
When running a livecheck on a cask that uses the `extract_plist` strategy to fetch the version, `brew livecheck` displays the error `Directory not empty @ dir_s_rmdir`.

Example:

```bash
% brew livecheck --cask clix
Error: clix: Directory not empty @ dir_s_rmdir - /private/tmp/d20220618-90762-yqxw5a
```

**Comment by p-linnane** [comment]
I am unable to replicate this. Can you rerun with `--debug` and post the output?

**Reply by PythonCoderAS** [reply]
https://gist.github.com/82706c45cd07c9d1b320fd56127beaee

**Reply by p-linnane** [reply]
Can you provide the output of `brew doctor` & `brew config`? Are you running some kind of MDM or EDR tool?

**Reply by PythonCoderAS** [reply]
> Can you provide the output of `brew doctor` & `brew config`? Are you running some kind of MDM or EDR tool?

I am not running any MDM or EDR tool, this is my personal iMac.

- Brew doctor: https://gist.github.com/PythonCoderAS/c70b534bf4f2b5664a0fb8a05b132470
- Brew config: https://gist.github.com/PythonCoderAS/7749d4558514b8fa4d72c9a3957622b6

**Reply by p-linnane** [reply]
Is there a reason you need to remain on macOS 11.4 with the 12.5 CLT? Those would be the first things I'd recommend fixing other than running a `brew update-reset` to get back to a cleaner state.

**Reply by PythonCoderAS** [reply]
> Is there a reason you need to remain on macOS 11.4 with the 12.5 CLT?

During a minor software update the shutdown for the iMac broke in such a weird way that 50% of the time shutdown would fail and the computer would only "shut down" via a watchdog kernel panic. It finally fixed itself on 11.4, so I'm not risking any more software updates that could re-break the system. 11.4 is stable for me.

**Comment by gromgit** [comment]
Are you running an anti-virus or other system security software? If so, try disabling it and running the install again.

**Reply by PythonCoderAS** [reply]
I do not have any AV software.

**Comment by PythonCoderAS** [comment, marked as answer]
Upon further investigation it seems that `clix.zip` outputs files with permissions `dr-x------`/`-r-x------`. I can either recursively chown the output directory or use `rm -rf`.

**Reply by gromgit** [reply]
Just to be sure, what's the output of `umask`?

**Reply by PythonCoderAS** [reply]
```shell
% umask
022
```

**Reply by gromgit** [reply]
OK, so your umask isn't stripping the user-writable permissions bit. I'm now properly stumped.

**Reply by PythonCoderAS** [reply]
I mean I get the same permissions issues when I open the zip file with the default MacOS archiver. @p-linnane can you extract the zip file on your own system and show the permissions of the output directory (`ls -l`)?

**Reply by p-linnane** [reply]
@PythonCoderAS 
```✝  ~  ls -l CLIX 
total 64
dr-x------  3 p-linnane  staff     96 Dec 31  1969 CLIX.app
drwx------  9 p-linnane  staff    288 Dec 31  1969 CLIX_cmds
-r--------  1 p-linnane  staff   4845 Dec 31  1969 EULA.html
-rw-------  1 p-linnane  staff   2839 Dec 31  1969 READ FIRST.html
-r-x------  1 p-linnane  staff  13816 Dec 31  1969 appleclean
-r--------  1 p-linnane  staff   2214 Dec 31  1969 appleclean.txt
```

**Reply by PythonCoderAS** [reply]
Yea that seems to match what I've been getting. So why am I running into the permission error and you're not?

**Reply by PythonCoderAS** [reply]
I think a simple fix is to run a chown on a temporary directory before ultimately removing it. This could cause a slowdown though, so maybe it could be done only when running into errors.

**Reply by SMillerDev** [reply]
Seeing how a routine macOS update also stopped your mac from turning off I'm going to guess either your hardware or your macOS install is severely broken. I don't think this is an issue in brew that needs fixing.
